src/npyffi/array.rs

```diff
@@ -68,8 +68,14 @@ impl PyArrayAPI {
     impl_api![47; PyArray_Zero(arr: *mut PyArrayObject) -> *mut c_char];
     impl_api![48; PyArray_One(arr: *mut PyArrayObject) -> *mut c_char];
     impl_api![49; PyArray_CastToType(arr: *mut PyArrayObject, dtype: *mut PyArray_Descr, is_f_order: c_int) -> *mut PyObject];
+    #[cfg(all(feature = "numpy-1", not(feature = "numpy-2")))]
     impl_api![50; PyArray_CastTo(out: *mut PyArrayObject, mp: *mut PyArrayObject) -> c_int];
+    #[cfg(all(not(feature = "numpy-1"), feature = "numpy-2"))]
+    impl_api![50; PyArray_CopyInto(dst: *mut PyArrayObject, src: *mut PyArrayObject) -> c_int];
```

Comment:
Since builds with support for versions will be common, I think this does not really add that much compile-time safety. I would rather suggest we extend the `impl_api` macro to do a runtime version check for function which are only available in one or the other version, e.g.

```rust
impl_api![@npy1 50;  PyArray_CastTo(out: *mut PyArrayObject, mp: *mut PyArrayObject) -> c_int];
impl_api![@npy2 50; PyArray_CopyInto(dst: *mut PyArrayObject, src: *mut PyArrayObject) -> c_int];
```

Reply:
That sounds good! I'm working on an update to the PR.